src/users/users.service.ts

```diff
@@ -119,6 +122,14 @@ export class UsersService {
     user.active = false;
     await this.userRepository.update(id, user);
     // await this.userRepository.delete(id);
+
+    // Unassign parcels
+    const userParcels = await this.parcelsService.getParcelsByUserId(id);
+    const unassignParcelPromises = userParcels
+      .filter(parcel => parcel.parcelTrackingStatus === ParcelStatus.assigned) // Unassign assigned parcels only
+      .map(parcel => this.parcelsService.unassignParcel(parcel.id));
```

Comment:
a thing to notice - I'm not sure parcelService.unassignParcel does the right thing.
Currently it is being used when removing a parcel and what it does is going over all the parcel's tracking status and changes them to "ready" and sets the userId to null.. it doesn't make much sense even when the parcel is deleted.
This pretty much deletes the entire history of the parcel, so maybe we should work on changing this method implementation, to not clear the history which might be meaningful..

Reply:
@tkennediatt 
You are right,
we should not delete all the history, actually, we should fix it just to update a new row in the parcel tracking that the parcel is back to the ready state and not assigned (if we want we can add in the comment of the parcel tracking -"שליח הוסר מהחבילה"
Also please fix the logs in unassignedParcel to be more accurate not remove parcel bu unassign parcel.
After completing this task we can continue discussing fixing the logic in deleting parcel :-)

Reply:
I changed the logs, and unassignParcel will now just add a new record into the tracking table.
And yes, we should probably fix the delete parcel method, so it won't delete the parcel, but will keep it as "inactive" perhaps, like the user removal method does.